src/reducers/__test__/articles.test.js

```diff
@@ -0,0 +1,459 @@
+
+describe('Testing relatedArticles reducer:', () => {
+  describe('IF action cannot be recognized, ', () => {
+    it('SHOULD return initial state WHEN previous state is undefined', () => {
+      expect(
+        reducers.relatedArticles(undefined, mockActions.notReconized)
+      ).to.deep.equal(initialState)
+    })
+    it('SHOULD return previous state WHEN previous state exsist', () => {
+      expect(
+        reducers.relatedArticles(mockPreviousState_relatedArticles, mockActions.notReconized)
+      ).to.deep.equal(mockPreviousState_relatedArticles)
+    })
+  })
+
+  describe(`If action type is ${types.FETCH_RELATED_ARTICLES_REQUEST}`, () => {
+    const action = mockActions[types.FETCH_RELATED_ARTICLES_REQUEST]
+    it('SHOULD return expected state(on initial state) WHEN previous state is undefined', () => {
+      expect(
+        reducers.relatedArticles(undefined, action)
+      ).to.deep.equal({
+        [action.id]: {
+          isFetching: true,
+          error: null,
+          hasMore: false,
+          items: []
+        }
+      })
+    })
+    it('SHOULD return expected state(on previous state) WHEN previous state exist', () => {
+      expect(
+        reducers.relatedArticles(mockPreviousState_relatedArticles, action)
+      ).to.deep.equal(mockExpectedStateOnPre_relatedArticles.request)
+    })
+  })
+
+  describe(`If action type is ${types.FETCH_RELATED_ARTICLES_FAILURE}`, () => {
+    const action = mockActions[types.FETCH_RELATED_ARTICLES_FAILURE]
+    it('SHOULD return expected state(on initial state) WHEN previous state is undefined', () => {
+      expect(
+        reducers.relatedArticles(undefined, action)
+      ).to.deep.equal({
+        [action.id]: {
+          isFetching: false,
+          error: action.error,
+          lastUpdated: action.failedAt
+        }
+      })
+    })
+    it('SHOULD return expected state(on previous state) WHEN previous state exist', () => {
+      expect(
+        reducers.relatedArticles(mockPreviousState_relatedArticles, action)
+      ).to.deep.equal(mockExpectedStateOnPre_relatedArticles.failure)
+    })
+  })
+
+  describe(`If action type is ${types.FETCH_RELATED_ARTICLES_SUCCESS}`, () => {
+    const action = mockActions[types.FETCH_RELATED_ARTICLES_SUCCESS]
+    it('SHOULD return expected state(on initial state) WHEN previous state is undefined', () => {
+      expect(
+        reducers.relatedArticles(undefined, action)
+      ).to.deep.equal({
+        [action.id]: {
+          isFetching: false,
+          hasMore: false,
+          error: null,
+          items: action.relatedIds,
+          total: action.relatedIds.length,
+          lastUpdated: action.receivedAt
+        }
+      })
+    })
+    it('SHOULD return expected state(on previous state) WHEN previous state exist', () => {
+      expect(
+        reducers.relatedArticles(mockPreviousState_relatedArticles, action)
+      ).to.deep.equal(mockExpectedStateOnPre_relatedArticles.success)
+    })
+  })
+})
+
```

Comment:
You can separate the `mockActions` and `mockExpectedState` into different `describe`.